pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2086,6 +2086,9 @@ EOT
     network = self.network_3_1_2_5;
   });
 
+  # Fixes build failure on ncurses-6.3
+  ncurses = appendPatch ./patches/KEY_EVENT-ncurses-6.3.patch super.ncurses;
```

Comment:
Is there somewhere you can pull this patch from upstream or something?  Normally we try to avoid carrying patches in Nixpkgs if possible.

Also, should this be reported upstream?  If so, please report upstream and add a link in the comment here.

Reply:
I don't think there is a public repository or bug tracker for `hackage:ncurses`. Emailed John (former `hackage:ncurses` maintainer) and Tom (current `hackage:ncurses` maintainer) to see if we have any place to report it upstream.

Reply:
Hmm, that's unfortunate.

Are you using this Haskell `ncurses` package somewhere?  If so, it seems like forking it and taking it over on Hackage might be a better idea than carrying around a patch here in Nixpkgs:

- There haven't been any releases in 5 years.
- The upstream repo seems dead or gone.
- There hasn't even been a revision in 4 years.
- There aren't any (important) reverse deps: https://packdeps.haskellers.com/reverse/ncurses

Since it doesn't seem to have any users, would anyone really benefit from only fixing it here in Nixpkgs?  If there is someone who would benefit, wouldn't it be better to take it over on Hackage so that it could benefit the whole Hackage community?

Reply:
Oh, maybe you're not a Haskeller, but you were just fixing this because of the fallout from https://discourse.nixos.org/t/uncoming-ncurses-6-2-6-3-update/16169?  In my opinion, it seems like you can just ignore `haskellPackages.ncurses` for now, since it seems dead.

Reply:
I personally don't use `hackage:ncurses` and would prefer not to take over upstream maintenance. Tom (current `hackage:ncurses` maintainer) said he will try to upload fixed version on hackage soon (maybe within weeks).

I'd also prefer not to introduce known regressions into `nixpkgs` to save peoples' time on investigation. Would `meta.broken = true` be a sensible use for known-broken packages?

Reply:
> Tom (current hackage:ncurses maintainer) said he will try to upload fixed version on hackage soon (maybe within weeks)

Oh, nice.  How did you know that Haskell ncurses had a new owner?  Looking at https://hackage.haskell.org/package/ncurses, I can't find anywhere that mentions a new owner.

> Would meta.broken = true be a sensible use for known-broken packages?

Yeah, that seems reasonable to me.  You could add `ncurses` to the package list in `pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml` and then re-run `maintainers/scripts/haskell/regenerate-hackage-packages.sh`.  Ideally after rebasing this on the `haskell-updates` branch.

Reply:
> Oh, nice. How did you know that Haskell ncurses had a new owner? Looking at https://hackage.haskell.org/package/ncurses, I can't find anywhere that mentions a new owner.

I asked John in private email (by sending a patch :) and he redirected me to Tom. Tom would like to cut a new release soon.

> Yeah, that seems reasonable to me. You could add ncurses to the package list in pkgs/development/haskell-modules/configuration-hackage2nix/broken.yaml and then re-run maintainers/scripts/haskell/regenerate-hackage-packages.sh. Ideally after rebasing this on the haskell-updates branch.

Aha, sounds reasonable. I'll fall back to it after a few days.

Reply:
Let's declare it a timeout. Marked `haskellPackages.ncurses` and it's two revdeps as broken.